creds/ssm/ssm.go

```diff
@@ -0,0 +1,75 @@
+package ssm
+
+import (
+	"bytes"
+	"strings"
+	"text/template"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/aws/aws-sdk-go/service/ssm/ssmiface"
+	varTemplate "github.com/cloudfoundry/bosh-cli/director/template"
+)
+
+type Ssm struct {
+	api            ssmiface.SSMAPI
+	teamName       string
+	pipelineName   string
+	secretTemplate *template.Template
+}
+
+func NewSsm(api ssmiface.SSMAPI, teamName string, pipelineName string, secretTemplate *template.Template) *Ssm {
+	return &Ssm{
+		api:            api,
+		teamName:       teamName,
+		pipelineName:   pipelineName,
+		secretTemplate: secretTemplate,
+	}
+}
+
+func (s *Ssm) buildSecretName(varName string) (string, error) {
+	var buf bytes.Buffer
+	err := s.secretTemplate.Execute(&buf, &SsmSecret{
+		Team:     s.teamName,
+		Pipeline: s.pipelineName,
+		Secret:   varName,
+	})
+	return buf.String(), err
+}
+
+func (s *Ssm) Get(varDef varTemplate.VariableDefinition) (interface{}, bool, error) {
+	secretName, err := s.buildSecretName(varDef.Name)
```

Comment:
Just noticed. It seems this implementation will only look at `/<team>/<pipeline>/<secret>` and not also attempt `/<team>/<secret>` if `<secret>` is not found for the specific pipeline. This contrasts with how the Vault backend  is implemented (see [here](https://github.com/concourse/atc/blob/master/creds/vault/vault.go#L30-L35)), and I think for consistency (and convenience) the SSM backend should work the same way.

Reply:
I shall add a `fallback-secret-template` that defaults to this path.